app/models/address.rb

```diff
@@ -98,7 +98,7 @@ def search(_location, components, postcode)
       address = Address.find_by('lower(street_address) IN (?)', components.map(&:downcase))
       return address if address
 
-      ukpc = UKPostcode.parse(postcode)
+      ukpc = UKPostcode.parse(postcode.to_s)
```

Comment:
this isn't javascript! `nil.to_s = ""` in ruby

Reply:
So it is. For the record, JS doesn't let you convert a nil value to anything else (which feels much more sensible).

I still think this is misbehaviour. Why is an empty string better than returning `nil`? Isn't a `nil` value a clearer indicator of emptiness?

Reply:
In JS this would be like `null || ""` which you can do in ruby as well. It's fairly common in ruby to cast things you are expecting to be strings explicitly. 

The whole importer is very complex with a lot of steps that rely on certain things being certain ways. (just look at the EventResolver). As originally specced events required addresses so any value that failed to resolve was skipped. This may have changed with Online Events but I haven't looked too closely at how that works.

But you are right in that this could do with being looked at and rationalised.